### PR TITLE
Added support for TraceEnable directive in vhosts.pp manifest

### PR DESCRIPTION
**Pull request by bschonec**
Since TraceEnable directive can be put in either the Apache server config or virtual host config, it makes sense to be able to have Puppet manage the directive in either place.

### CODE REVIEW

**Comment by bschonec** [comment]
Added ability to override Apache main configuration file (typically httpd.conf) variable $conf_file.  This give the user the ability to manage multiple configuration files on a single servers.  This is advantages if the server is running both PHP4 and PHP5.  An example would be to have a 'httpd.conf.php4' and 'httpd.conf.php5' to distinguish between the two concurrent installs of PHP.

File: manifests/mod/php4.pp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  validate_array($extensions)
+
+  if $source and ($content or $template != 'apache/mod/php4.conf.erb') {
+    warning('source and content or template parameters are provided. source parameter will be used')
+  } elsif $content and $template != 'apache/mod/php4.conf.erb' {
+    warning('content and template parameters are provided. content parameter will be used')
+  }
+
+  $manage_content = $source ? {
+    undef   => $content ? {
```
Comment (bmjen) [inline manifests/mod/php4.pp]: Needs Alignment

File: manifests/init.pp [hunk 2]
```diff
@@ -66,10 +68,12 @@
   $apache_version         = $::apache::version::default,
   $server_tokens          = 'OS',
   $server_signature       = 'On',
-  $trace_enable           = 'On',
+  $trace_enable           = undef,
```
Comment (cyberious) [inline manifests/init.pp]: Why are we changing the default behavior, this is a breaking change.

File: manifests/init.pp [hunk 3]
```diff
@@ -244,11 +263,12 @@
...
   }
 
-  if $::apache::conf_dir and $::apache::params::conf_file {
+if $::apache::manage_conf_file {
+  if $::apache::conf_dir and $::apache::conf_file {
     case $::osfamily {
       'debian': {
         $pidfile              = "\${APACHE_PID_FILE}"
-        $error_log            = 'error.log'
+        $error_log            = 'error_log'
```
Comment (cyberious) [inline manifests/init.pp]: Another breaking change. Why?

File: manifests/mod/php4.pp [hunk 4]
```diff
@@ -0,0 +1,62 @@
+class apache::mod::php4 (
```
Comment (cyberious) [inline manifests/mod/php4.pp]: This is a good idea, but should be a separate PR

File: templates/mod/ssl.conf.erb [hunk 5]
```diff
@@ -9,17 +9,17 @@
 
   SSLPassPhraseDialog <%= @ssl_pass_phrase_dialog %>
   SSLSessionCache "shmcb:<%= @session_cache %>"
-  SSLSessionCacheTimeout <%= @ssl_sessioncachetimeout %>
+  SSLSessionCacheTimeout 300
```
Comment (tphoney) [inline templates/mod/ssl.conf.erb]: why are you hardcoding these values, they were previously variables

File: checksums.json [hunk 6]
```diff
@@ -0,0 +1,289 @@
+{
```
Comment (jonnytpuppet) [inline checksums.json]: This file should not be commited

File: manifests/init.pp (hunk 2 above)
Comment (jonnytpuppet) [inline manifests/init.pp]: Would this cause a backwards incompatibility?

**Comment by bschonec** [comment]
I'm quite new at all of this and I apologize for the 'noise' in the code.  I'll pull the latest version and then clean up and resubmit.

**Comment by igalic** [comment]
@bschonec: no worries about noise!
